linera-views/src/backends/rocks_db.rs

```diff
@@ -45,21 +44,210 @@ const MAX_KEY_SIZE: usize = 8388208;
 /// The RocksDB client that we use.
 pub type DB = rocksdb::DBWithThreadMode<rocksdb::MultiThreaded>;
 
+/// The choice of the spawning mode.
+/// The SpawnBlocking always works and is the safest.
+/// The BlockInPlace can only be used in multi-threaded environment.
+/// One way to select that is to select BlockInPlace when
+/// tokio::runtime::Handle::current().metrics().num_workers() > 1
+/// The BlockInPlace is documented in <https://docs.rs/tokio/latest/tokio/task/fn.block_in_place.html>
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum RocksDbSpawnMode {
+    /// This uses the `spawn_blocking` function of tokio.
+    SpawnBlocking,
+    /// This uses the `block_in_place` function of tokio.
+    BlockInPlace,
+}
+
+/// Obtains the spawning mode from runtime
+pub fn get_spawn_mode_from_runtime() -> RocksDbSpawnMode {
```

Comment:
Can this be a static method of `RocksDbSpawnMode`? thanks

Reply:
Yes, done.